src/app/reporting/Engine.cpp

```diff
@@ -57,11 +57,27 @@ Engine::RetrieveClusterData(AttributeDataElement::Builder & aAttributeDataElemen
     CHIP_ERROR err                              = CHIP_NO_ERROR;
     TLV::TLVType type                           = TLV::kTLVType_NotSpecified;
     AttributePath::Builder attributePathBuilder = aAttributeDataElementBuilder.CreateAttributePathBuilder();
-    attributePathBuilder.NodeId(aClusterInfo.mNodeId)
-        .EndpointId(aClusterInfo.mEndpointId)
-        .ClusterId(aClusterInfo.mClusterId)
-        .FieldId(aClusterInfo.mFieldId)
-        .EndOfAttributePath();
+    attributePathBuilder.NodeId(aClusterInfo.mNodeId).EndpointId(aClusterInfo.mEndpointId).ClusterId(aClusterInfo.mClusterId);
+    AttributePathSelector * current = aClusterInfo.mpAttributePathSelector;
+
+    while (current != nullptr)
+    {
+        if (current->mFlag == AttributePathSelectorFlag::kFieldIdValid)
```

Comment:
pleas add a flag validity here as well.

if flags are mutually exclusive, please use an enum not a bitFlag. If flags are not mutually exclusive, update this code and the code before to handle them as such.